tests/integration/test_charm.py

```diff
@@ -378,20 +396,36 @@ async def test_seldon_predictor_server(ops_test: OpsTest, server_config, url, re
 
     assert sorted(response.items()) == sorted(resp_data.items())
 
-    client.delete(seldon_deployment, name=model, namespace=namespace)
-    client.delete(Service, name=service_name, namespace=namespace)
+    client.delete(seldon_deployment, name=ml_model, namespace=namespace, grace_period=0)
+
+    # wait for application to settle
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60, idle_period=30
+    )
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=2, min=1, max=10),
+    stop=tenacity.stop_after_attempt(60),
+    reraise=True,
+)
+async def assert_removed(ops_test: OpsTest):
+    """Remove application and verify it is removed."""
+    logger.info(f"Removing application {APP_NAME}")
+    # TO-DO: use this: await ops_test.run("juju", "remove-application", f"{APP_NAME}")
+    subprocess.run(["juju", "remove-application", f"{APP_NAME}"])
```

Comment:
Having `subprocess.run` inside this `tenacity.retry` means we'll make a new `juju remove-application` call every retry - is that what is intended?  Alternatively, `juju remove-application` could be separate so it only gets called once, then the `assert_removed` repeatedly checks to see if things have been removed.

Reply:
I would like to issue multiple `juju remove-application`. Just wanted to ensure that the event is actually gets to the charm.
When we figure out what is wrong with `mode.remove_application()` and/or what's causing the race condition, then I will switch it back to it.